File: lib/app/services/providers/ion_connect/shareable_identifier.dart
// SPDX-License-Identifier: ice License 1.0

import 'package:ion/app/services/providers/ion_connect/ion_connect_protocol_identifier_type.dart';

class ShareableIdentifier {
  ShareableIdentifier({
    required this.prefix,
    required this.special,
    this.relays = const [],
    this.author,
    this.kind,
  });
  final IonConnectProtocolIdentifierType prefix;
  final String special;
  List<String> relays;
  String? author;
  int? kind;
}

File: ios/NotificationServiceExtension/Translation/Models/ShareableIdentifier.swift
// SPDX-License-Identifier: ice License 1.0

import Foundation

struct ShareableIdentifier {
    let prefix: String
    let data: [String: String]
    
    init(prefix: String, data: [String: String]) {
        self.prefix = prefix
        self.data = data
    }
}
